### PR TITLE
[15.0][ADD] web_widget_remote_measure: New module

### PR DESCRIPTION
**Pull request by chienandalu**
cc @Tecnativa TT38745

ping @carlosdauden 

Example of behavior in the oncoming `stock_remote_device`:

remote-device-example

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Being web client implementation, I think it should be called `web_widget_remote_measure` or similar.

**Comment by chienandalu** [comment]
Ok, changed :)

File: web_widget_remote_measure/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1,5 @@
+To configure your remote devices:
+
+#. Go to *Settings > Technichal > Resource > Remote devices*
```
Comment (legalsylvain) [inline web_widget_remote_measure/readme/CONFIGURE.rst]:
```suggestion
#. Go to *Settings > Technical > Resource > Remote devices*
```

File: web_widget_remote_measure/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1,5 @@
+This module allows to input data from remote decives in your network. Currently, only
```
Comment (legalsylvain) [inline web_widget_remote_measure/readme/DESCRIPTION.rst]:
```suggestion
This module allows to input data from remote devices in your network. Currently, only
```

File: web_widget_remote_measure/readme/USAGE.rst [hunk 3]
```diff
@@ -0,0 +1,15 @@
+The remote device has to be in the users network so their web clients can reach them.
+
+In order to test a device you can:
+
+#. Go to *Settings > Technichal > Resource > Remote devices*
```
Comment (legalsylvain) [inline web_widget_remote_measure/readme/USAGE.rst]:
```suggestion
#. Go to *Settings > Technical > Resource > Remote devices*
```

**Comment by pedrobaeza** [comment]
I wonder the overlap with https://odoo-community.org/shop/remote-base-3251?search=remote#attr=21539 and indeed seeing this code, it's including in the same module the base definition with the widget.

**Comment by pedrobaeza** [comment]
This is also a very specific "remote" thing, so maybe including `websocket` reference or something can be convenient.

**Comment by chienandalu** [comment]
> I wonder the overlap with https://odoo-community.org/shop/remote-base-3251?search=remote#attr=21539 and indeed seeing this code, it's including in the same module the base definition with the widget.

I'd say that module's purpose is to store the accesses to the Odoo server

**Comment by chienandalu** [comment]
> This is also a very specific "remote" thing, so maybe including `websocket` reference or something can be convenient.

It can be extended to a Webservices interface in the future if needed

**Comment by pedrobaeza** [comment]
Then `remote_webservice` prefix?

**Comment by chienandalu** [comment]
But it could be either webservices or websockets

**Comment by pedrobaeza** [comment]
OK, I don't insist then.

File: web_widget_remote_measure/readme/CONFIGURE.rst (hunk 1 above)
Comment (chienandalu) [reply web_widget_remote_measure/readme/CONFIGURE.rst]: Thanks :)

**Comment by rousseldenis** [comment]
And why not envisaging connecting devices like IOT (that could be connected to POS) and more specifically to pywebdriver ?

That allows you to implement protocols there. With the advantage of being available for all flows you want.

**Comment by chienandalu** [comment]
> And why not envisaging connecting devices like IOT (that could be connected to POS) and more specifically to pywebdriver ?
> That allows you to implement protocols there. With the advantage of being available for all flows you want.

Connection to any remote measure device (IoT or the device itself via web services) would be possible if implemented but we want to keep the infrastructure required to implement this in the customer's setup as simple as possible.

For our current approach, adding an IoT (or pywebdriver) in the middle is an extra point of potential failure that can complicate our maintenance.

**Comment by rousseldenis** [comment]
> > And why not envisaging connecting devices like IOT (that could be connected to POS) and more specifically to pywebdriver ?
> > That allows you to implement protocols there. With the advantage of being available for all flows you want.
> 
> Connection to any remote measure device (IoT or the device itself via web services) would be possible if implemented but we want to keep the infrastructure required to implement this in the customer's setup as simple as possible.
> 
> For our current approach, adding an IoT (or pywebdriver) in the middle is an extra point of potential failure that can complicate our maintenance.

So, implementing a particular protocol should maybe go in another module

**Comment by bosd** [comment]
Let us know when you're ready

**Review by carlosdauden — APPROVED** [review]

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**